### ISSUE TITLE
Import actions within the `Import/Export` collections modal need a loading state

### ISSUE THREAD

**Issue by jamesgeorge007** [hoppscotch/hoppscotch#4145]
> [!NOTE]
This is intended for [FOSSHACK](https://github.com/hoppscotch/hoppscotch/discussions/4131) participants.

### Description

While importing a large collection in a team workspace, it'll be nice to have a loading state for the `Import` button while the network calls happen behind the scenes.

  https://github.com/hoppscotch/hoppscotch/assets/25279263/0c773ddb-c0c6-4e3d-aed4-679b2a8a318f

> It should also stay disabled to prevent creating duplicate entries if clicking `Import` multiple times.

**Comment by kyoichishido** [comment]
Hi, I wanna work on this, please assign me. I can't find the guide to set-up this repo locally

**Comment by jamesgeorge007** [comment]
Hi @kyoichishido, sorry for the delayed response. To reiterate, this is intended for [FOSSHACK ](https://fossunited.org/fosshack/2024)participants. Please make sure to register as a participant if you haven't already. You can find more information [here](https://github.com/hoppscotch/hoppscotch/discussions/4131).

Follow the below steps to set up the repo locally.

- Please ensure the prerequisites are met by following the guide [here](https://docs.hoppscotch.io/documentation/self-host/community-edition/prerequisites#install-node-js-npm-pnpm). The necessary steps to create OAuth apps to enable third-party authentication can be found [below](https://docs.hoppscotch.io/documentation/self-host/community-edition/prerequisites#oauth).
- Clone the repo locally and navigate to the path.
- Create a `.env` file at the project root copying over the contents from [.env.example](https://github.com/hoppscotch/hoppscotch/blob/main/.env.example). More information can be found [here](https://docs.hoppscotch.io/documentation/self-host/community-edition/install-and-build#configuring-the-environment).
- The codebase is a monorepo based on [pnpm workspaces](https://pnpm.io/workspaces). Run `pnpm i` at the project root that will install the dependencies followed by running the [postinstall script](https://docs.npmjs.com/cli/v10/using-npm/scripts#pre--post-scripts) for the applicable packages.
- Run `pnpm dev` from the same path that spins up the dev server.

> Backend setup

- Build the backend container via `docker compose build hoppscotch-backend`.
- Now run `docker compose run --entrypoint sh hoppscotch-backend`.
- The above command opens a shell inside an instance of the backend container. Now run `pnpm prisma migrate deploy` to run the migrations. More information can be found [here](https://docs.hoppscotch.io/documentation/self-host/community-edition/install-and-build#running-migrations).
- Spin up the backend container via ` docker compose up hoppscotch-backend`.

Alternatively, you can use the [AIO container](https://docs.hoppscotch.io/documentation/self-host/community-edition/install-and-build#using-the-aio-container) which is a single container that provides all the services required to run Hoppscotch.

P.S: Any updates locally will require rebuilding the AIO container. For development, the above approach where the FE dev server is spun up locally along with the backend container would suffice.

```
# Build the container locally
docker compose build hoppscotch-aio

# Run migrations
docker compose run --entry-point hoppscotch-aio

# While at the launched shell with the above command
pnpm prisma migrate deploy

# Followed by
docker compose up hoppscotch-aio
```

Let us know if you have any questions.

**Comment by HelixY2J** [comment]
I am participating in FOSS Hack where I have registered and joined a team. My initial approach for this is to set a flag that will disable the button and show a loading message while the async process is running. Once the process completes, the flag will be toggled back.Can this issue be assigned to me?

**Comment by jamesgeorge007** [comment]
Hi @HelixY2J, thanks for expressing interest. @kyoichishido, can you confirm if you're interested in picking this up? Else, this can be assigned to @HelixY2J.

**Comment by jamesgeorge007** [comment]
Assigning this to @HelixY2J.

**Comment by HelixY2J** [comment]
I created a reactive variable "isImporterInProgress "using Vue ref function and added it into the existing import logic in collections/importexport.vue. It will be set  true when the import starts and reset to false after completion.Can I get some feedback on whether this approach is appropriate or there are any alternative methods that should be considered for handling the loading state

**Comment by jamesgeorge007** [comment]
Hi, a quick suggestion regarding the behaviour. It would be nice to have a loading spinner before the button label `Import` while the button stays disabled.

**Comment by jamesgeorge007** [comment]
Implemented in #4217 and released as part of v2024.9.0.
